src/contexts/Maci.tsx

```diff
@@ -84,10 +84,33 @@ export const MaciProvider: React.FC<MaciProviderProps> = ({ children }: MaciProv
     setMaciPubKey(userKeyPair.publicKey);
   }, [address, signatureMessage, signMessageAsync, setMaciPrivKey, setMaciPubKey]);
 
-  const votingEndsAt = useMemo(
-    () => (pollData ? new Date(Number(pollData.deployTime) * 1000 + Number(pollData.duration) * 1000) : new Date()),
-    [pollData],
-  );
+  const isDifferentPoll = (a: IGetPollData, b: IGetPollData): boolean => {
+    if (a.id !== b.id) {
+      return true;
+    }
+
+    if (a.address !== b.address) {
+      return true;
+    }
+
+    if (a.deployTime !== b.deployTime) {
+      return true;
+    }
+
+    if (a.duration !== b.duration) {
+      return true;
+    }
+
+    if (a.numSignups !== b.numSignups) {
+      return true;
+    }
+
+    if (a.isStateAqMerged !== b.isStateAqMerged) {
+      return true;
+    }
+
+    return false;
```

Comment:
In the function no need to use if for every case.
```suggestion
    return a.address !== b.address || a.deployTime !== b.deployTime || a.duration !== b.duration || a.numSignups !== b.numSignups || a.isStateAqMerged !== b.isStateAqMerged;
```
And what's about using `isEqual#lodash` and `pick#lodash` specific fields for checking the data?

Reply:
change to this: https://github.com/Ethereum-Mex/ethmx-maci-rpgf/blob/development/src/contexts/Maci.tsx#L99-L108